apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -11,7 +11,7 @@
 from .options import SOMATileDBContext
 
 
-def build_index(
+def build_index_tiledb_context(
```

Comment:
This is a good idea. Might also make sense to make `context` and `thread_count` mutually exclusive, though your suggested behavior of "use thread count if context is not present otherwise do not care about it" is also reasonable.

Reply:
Sounds good to me.